File: problem-001/nim/sumbyto.nim
proc sumbyto(start: int, by: int, to: int): int =
    if start > to:
        result = 0
    else:
        result = start + sumbyto(start + by, by, to)

var val = sumbyto(0, 5, 999) + sumbyto(0, 3, 999) - sumbyto(0, 15, 999)
echo(val)

File: problem-001/elixir/sumbyto.exs
defmodule P001 do
  def sumbyto(start, by, to) do
    if start > to do
      0
    else
      start + (sumbyto(start + by, by, to))
    end
  end
end

IO.puts(P001.sumbyto(0, 3, 999) + P001.sumbyto(0, 5, 999) - P001.sumbyto(0, 15, 999))
